utils/src/pickTreeItem/experiences/compatibility/PickTreeItemWithCompatibility.ts

```diff
@@ -28,4 +35,37 @@ export namespace PickTreeItemWithCompatibility {
 
         return createSubscriptionContext(applicationSubscription);
     }
+
+    /**
+     * Helper to provide compatibility for `AzExtParentTreeItem.showTreeItemPicker`.
+     */
+    export async function showTreeItemPicker<TPick extends types.AzExtTreeItem>(context: types.ITreeItemPickerContext, tdp: vscode.TreeDataProvider<ResourceGroupsItem>, expectedContextValues: string | RegExp | (string | RegExp)[], startingTreeItem?: AzExtTreeItem): Promise<TPick> {
+        const promptSteps: AzureWizardPromptStep<types.QuickPickWizardContext>[] = [
+            new CompatibilityRecursiveQuickPickStep(tdp, {
+                contextValueFilter: {
+                    include: expectedContextValues,
+                },
+                skipIfOne: false,
+            }),
+        ];
+
+        // Fill in the `pickedNodes` property
+        const wizardContext = { ...context } as types.QuickPickWizardContext;
```

Comment:
Should this code not be moved into its own experience?

Reply:
Though initially I had the same thought as you, there is no v2 equivalent of showTreeItemPicker so there doesn't need to be an experience for it. The other functions in here are wrapping experiences with some compatibility stuff. I think it's fine to keep the code here.